PLEP-0007.rst

```diff
@@ -0,0 +1,847 @@
+element, and spectral methods. If possible, the code for the
+numerical method will be generated from the specified equations
+(though this may require that the equations be specified in
+conservative form).
+
+If the users choose a PIC simulation, then they will choose the
+particle pusher for the time advance.
+
+At this point, users will specify the numerical input parameters.
+
+.. Right now I am not certain how to deal with boundary conditions, and
+   how we could treat them similarly for fluid and PIC simulations.
+   If we have a problem setup for a fluid case that has Dirichlet BCs
+   for density, then how do we transfer that to a PIC case?  Should we
+   have BCs defined separately?  Or include them in the numerical method
+   definition maybe?  Maybe we should have a way to convert a fluid
+   simulation setup into a PIC simulation.
+
+.. Thinking more: if the boundary conditions are different between
+   fluid and PIC, then they should be denoted in different ways.  We
+   could provide some suggestions on adapting a problem for PIC vs
+   fluid approximation simulations.
+
+.. Users will next define the system of equations or physical model to be
+   solved. It is at this point that users will choose the style of
+   simulation (including but not limited to fluid, particle-in-cell, and
+   hybrid approaches). The physical model will be checked to be consistent
+   with the initial and boundary conditions.
+
+.. Users define the physical model.
+     Options for user input
+      Pre-defined sets of equations with options to specify different
+      coefficients (like resistive MHD with uniform, Spitzer, anomalous,
+      or a user-defined function)
+    List of strings containing the different equations
+   If the equations are in conservative form (including with sources
+    and sinks) then
+   We can have pre-defined sets of equations
+   We can have pre-defined sets of equations as a string (including
+   unicode characters) following Dedalus approach
+    This can be done best if numerical method can be automagically generated
+    Julia allows us to pass functions around as arguments (A function
+      can have a function as an argument, and a function can return a
+      function)
+   We can sometimes use pre-set equations (like resistive MHD, with
+     uniform or temperature dependent or anomalous resistivity)
+     Numerical method
+     Post-processing
+     Maybe we could create a function that automatically writes text that
+       describes the numerical method and such.
+
+.. Grid generation
+   ---------------
+
+.. More detail needed on grid generation. Need to discuss mesh
+   packing capabilities and how to generate complicated grids. For
+   finite element simulations, more information on the mapping will be
+   necessary, but might not be worth discussing here.
+
+.. Users will be able to generate the grid after the domain is
+   specified and the numerical method is chosen.
```

Comment:
People might also want to load grids from existing geometry generators, say, field line setups in tokamaks... Out of scope for now?

Reply:
We're taking the Borg approach for plasma simulation code development, so it will be in scope eventually!